app/controllers/self_service/firms_controller.rb

```diff
@@ -7,5 +7,13 @@ def index
         @trading_names.map(&:registered_name).include? lookup_name.name
       end
     end
+
+    def destroy
+      trading_name = principal.firm.trading_names.registered.find(params[:id])
+      trading_name.destroy
```

Comment:
The principal must exist, and they must have a firm.
The trading names could be empty, then the registered scope would have nothing to operate on which would be fine.

The `.find` would get upset if there was no trading name with that id.
The only way I can see that scenario is if someone if hacking the html src, and it seems reasonable for active record to throw an exception in that situation.

Is there something in particular you're thinking of?